### PR TITLE
Added order PATCH capability

### PR DESCRIPTION
**Pull request by mb-ca**
Added ability to patch an order and any of its child properties (i.e. customer, shipping address).  `patch()` is similar to the `update()` method, except it accepts a partial order object.

No known drawbacks.

### CODE REVIEW

File: e2e/ChannelApeClient.spec.ts [hunk 1]
```diff
@@ -684,4 +716,32 @@ describe('ChannelApe Client', () => {
...
+  function randomNumber(min: number, max: number): number {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  }
+
+  /**
+   * Generates a random semi-readable word based on the provided min an max length values.
+   * @param {number} min Minium word length (inclusive)
+   * @param {number} max Maximum word length (inclusive)
+   */
+  function randomWord(min: number, max: number): string {
```
Comment (rjdavis3) [inline e2e/ChannelApeClient.spec.ts]: We already have fake word generator. Please see the order creation test in this class.

Please remove this `randomWord` function and replace calls to this function with calls to the `faker` sdk we already have in place.

File: e2e/ChannelApeClient.spec.ts [hunk 2]
```diff
@@ -684,4 +716,32 @@ describe('ChannelApe Client', () => {
     expect(channel.updatedAt.getUTCMilliseconds())
       .to.be.greaterThan(expectedCreatedAt.getUTCMilliseconds());
   }
+
+  /**
+   * Generates a random number based on the provided min and max values.
+   * @param {number} min Minimum number to generate (inclusive)
+   * @param {number} max Maximum number to generate (inclusive)
+   */
+  function randomNumber(min: number, max: number): number {
```
Comment (rjdavis3) [inline e2e/ChannelApeClient.spec.ts]: We already have fake word generator. Please see the order creation test in this class.

Please remove this `randonmNumber` function and replace calls to this function with calls to the `faker` sdk we already have in place.

File: src/orders/model/OrderPatchRequest.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import Order from './Order';
+import { RecursivePartial } from '../../model/RecursivePartial';
+
+export default interface OrderPatchRequest extends RecursivePartial<Order> {
+  actionId?: string;
```
Comment (rjdavis3) [inline src/orders/model/OrderPatchRequest.ts]: This is my fault for not bringing this up in code review but we need one more optional boolean field here titled `channelSync`. That is the field we will use to sync changes back to Shopify when true on the REST API.

Can you also add this in OrderUpdateRequest.ts as well?

File: e2e/ChannelApeClient.spec.ts (hunk 1 above)
Comment (mb-ca) [reply e2e/ChannelApeClient.spec.ts]: @rjdavis3 Didn't know about faker, so much better than what I had!

**Comment by mb-ca** [comment]
@rjdavis3 I was thinking that the customer or address classes had required properties which is why I created `RecursivePartial`, but just double checked and I was wrong.  The class was simple enough to remove so I did take it out.

**Pull request merged**